Source/WebKit/GPUProcess/graphics/ImageBufferShareableAllocator.cpp

```diff
@@ -32,20 +32,42 @@
 #include <WebCore/ImageBuffer.h>
 #include <wtf/TZoneMallocInlines.h>
 
+#if HAVE(IOSURFACE)
+#include <WebCore/ImageBufferIOSurfaceBackend.h>
+#endif
+
 #if ENABLE(GPU_PROCESS)
 
 namespace WebKit {
 using namespace WebCore;
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(ImageBufferShareableAllocator);
 
+#if HAVE(IOSURFACE)
+ImageBufferShareableAllocator::ImageBufferShareableAllocator(const ProcessIdentity& resourceOwner, IOSurfacePool* ioSurfacePool)
+    : m_resourceOwner(resourceOwner)
+    , m_ioSurfacePool(ioSurfacePool)
+{
+}
+#else
 ImageBufferShareableAllocator::ImageBufferShareableAllocator(const ProcessIdentity& resourceOwner)
     : m_resourceOwner(resourceOwner)
 {
 }
+#endif
 
-RefPtr<ImageBuffer> ImageBufferShareableAllocator::createImageBuffer(const FloatSize& size, const DestinationColorSpace& colorSpace, RenderingMode) const
+RefPtr<ImageBuffer> ImageBufferShareableAllocator::createImageBuffer(const FloatSize& size, const DestinationColorSpace& colorSpace, RenderingMode renderingMode) const
 {
+#if HAVE(IOSURFACE)
+    if (renderingMode == RenderingMode::Accelerated && m_ioSurfacePool) {
+        ImageBufferCreationContext creationContext;
+        creationContext.surfacePool = m_ioSurfacePool.get();
+        creationContext.resourceOwner = m_resourceOwner;
+
+        return ImageBuffer::create<ImageBufferIOSurfaceBackend>(size, 1, colorSpace, { PixelFormat::BGRA8 }, RenderingPurpose::Unspecified, creationContext);
+    }
+#endif
+
     RefPtr<ImageBuffer> imageBuffer = ImageBuffer::create<ImageBufferShareableBitmapBackend>(size, 1, colorSpace, { PixelFormat::BGRA8 }, RenderingPurpose::Unspecified, { });
```

Comment:
If RenderingMode is `Accelerated` and you don't have an m_ioSurfacePool, you still should make an accelerated buffer.